### DISCUSSION TITLE
SemVer ABI break in detail headers from v3.7.3 and v3.10.5

### DISCUSSION THREAD

**Discussion by Ryanf55** [nlohmann/json#4135] [General]
Between Ubuntu 20, and Ubuntu 22, I am facing compile errors with due to changes in the public header.

https://github.com/nlohmann/json/blob/4f8fba14066156b73f1189a2b8bd568bde5284c5/include/nlohmann/detail/exceptions.hpp#L127-L128
https://github.com/nlohmann/json/blob/v3.7.3/include/nlohmann/detail/exceptions.hpp#L129

Is it intended that all the public headerfiles be SemVer compliant, or is the detail namespace excluded?

We only used the `nlohmann/json.hpp` as the include. 

Relates to https://github.com/nlohmann/json/discussions/3473#discussion-4058251.

**Comment by gregmarr** [comment]
Are you using something in the `detail` namespace directly?

**Reply by Ryanf55** [reply]
Yes. The above function call to `create(...)`

**Reply by falbrechtskirchinger** [reply]
The `detail` namespace isn't part of the public API. Use at your own risk. Note, that the `create()` functions changed again in v3.11.x (`context` was changed to a pointer argument). (Since you mentioned ABI, there were zero guarantees about ABI compatibility before v3.11.x, i.e., all code sharing objects has to be compiled against the same version of the library.)
